resources/src/commands/ping.rs

```diff
@@ -6,5 +6,6 @@ use crate::slack;
 // the Slack bot.
 pub async fn run(channel: &str) {
     let text = "pong";
-    let _status = slack::chat::post_message(text, channel, Option::None).await;
+    let _status_with_bot_token = slack::chat::post_message(&text, &channel, true).await;
+    let _status_with_user_token = slack::chat::post_message(text, channel, false).await;
```

Comment:
Can you not pass the reference (e.g. `&text`, `&channel`)?

Reply:
Pass the reference twice? It's deallocated after the second use. Is there a benefit to having the function who borrowed it clean up the memory vs. Passing it back to the original owner to clean it up?

Reply:
You're `awaiting` the first function, passing the reference to the second function is fine as they would never be modified at the same time - also you're passing immutable references so they're read only. Generally you should always pass-by-reference if the values are read only since passing-by-value means a copy is made.
Either pass-by-value in both functions or pass-by-reference, treating these two functions asymmetrically could cause bugs in the future.